server.go

```diff
@@ -936,3 +756,280 @@ func (s *Server) handlerHistory(w http.ResponseWriter, r *http.Request) {
+		s.doError(
+			http.StatusInternalServerError,
+			"something went wrong :C",
+			w, r)
+		return
+	}
+
+	if len(remarkstext) > maxRemarksLength {
+		s.l.Debug("Remarks too long: %d", len(remarkstext))
+		data.ErrorMessage = append(data.ErrorMessage, fmt.Sprintf("Remarks too long! Max Length: %d characters", maxRemarksLength))
+		data.ErrRemarks = true
+	}
+	// Here we continue with the other input checks
+	maxTitleLength, err := s.data.GetCfgInt(ConfigMaxTitleLength, DefaultMaxTitleLength)
+	if err != nil {
+		s.l.Error("Unable to get %q: %v", ConfigMaxTitleLength, err)
+		s.doError(
+			http.StatusInternalServerError,
+			"something went wrong :C",
+			w, r)
+		return
+	}
+
+	title := strings.TrimSpace(r.FormValue("MovieName"))
+	data.ValTitle = title
+
+	if data.ValTitle == "" {
+		data.ErrorMessage = append(data.ErrorMessage, "Missing movie title")
+		data.ErrTitle = true
+	}
+
+	if len(data.ValTitle) > maxTitleLength {
+		s.l.Debug("Title too long: %d", len(data.ValTitle))
+		data.ErrTitle = true
+		data.ErrorMessage = append(data.ErrorMessage, fmt.Sprintf("Title too long! Max Length: %d characters", maxTitleLength))
+	} else if len(data.ValTitle) == 0 {
+		s.l.Debug("Title too short: %d", len(common.CleanMovieName(data.ValTitle)))
+		data.ErrTitle = true
+		data.ErrorMessage = append(data.ErrorMessage, fmt.Sprintf("Title too short! Min Length: %d characters", 1))
+	}
+
+	movieExists, err := s.data.CheckMovieExists(title)
+	if err != nil {
+		s.doError(
+			http.StatusInternalServerError,
+			fmt.Sprintf("Unable to check if movie exists: %v", err),
+			w, r)
+		return
+	}
+
+	if movieExists {
+		data.ErrTitle = true
+		s.l.Debug("Movie exists")
+		data.ErrorMessage = append(data.ErrorMessage, "Movie already exists")
+	}
+
+	descr := strings.TrimSpace(r.FormValue("Description"))
+	data.ValDescription = descr
+
+	maxDescriptionLength, err := s.data.GetCfgInt(ConfigMaxDescriptionLength, DefaultMaxDescriptionLength)
+	if err != nil {
+		s.l.Error("Unable to get %q: %v", ConfigMaxTitleLength, err)
+		s.doError(
+			http.StatusInternalServerError,
+			"something went wrong :C",
+			w, r)
+		return
+	}
+
+	if len(data.ValDescription) > maxDescriptionLength {
+		s.l.Debug("Description too long: %d", len(data.ValDescription))
+		data.ErrDescription = true
+		data.ErrorMessage = append(data.ErrorMessage, "Description too long!")
+	}
+
+	if len(descr) == 0 {
+		data.ErrDescription = true
+		data.ErrorMessage = append(data.ErrorMessage, "Missing description")
+	}
+
+	var posterpath string
+
+	posterFileName := strings.TrimSpace(r.FormValue("MovieName"))
+
+	posterFile, _, err := r.FormFile("PosterFile")
+
+	if posterFile != nil {
+		if err != nil {
+			s.l.Debug("Formfile Error")
+			data.ErrPoster = true
+			data.ErrorMessage = append(data.ErrorMessage, err.Error())
+		}
+
+		file, err := s.uploadFile(r, posterFileName)
+
+		if err != nil {
+			data.ErrPoster = true
+			data.ErrorMessage = append(data.ErrorMessage, err.Error())
```

Comment:
This error should be logged.  Additionally, a generic error should probably be given to the user as the returned error can have system details.

Reply:
Will be changed in the next commit